embulk-core/src/main/java/org/embulk/spi/ExecSession.java

```diff
@@ -219,6 +233,11 @@ public boolean isPreview()
 
     public void cleanup()
     {
+        for (final Map.Entry<Reporter.Channel, Reporter> e : reporters.entrySet()) {
```

Comment:
Yea, the problem is that `reporters` is null because `EmbulkTestRuntime` 
 doesn't set reporter objects to the `reporters` in the session. Fixed 06dc5422d3b86cee8c5cba8ea7833d8f7f5d5f93